pkgsettings/pkgsettings.py

```diff
@@ -98,18 +107,23 @@ def __enter__(self) -> Self:
         self._override_enable()
         return self
 
-    def __exit__(self, exc_type, exc_value, traceback) -> None:
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_value: BaseException | None,
+        traceback: TracebackType | None,
+    ) -> None:
         self._override_disable()
 
     @overload
-    def __call__(self, func: _F) -> _F:
+    def __call__(self, func: Callable[P, T]) -> Callable[P, T]:
         ...
 
     @overload
-    def __call__(self, **override_settings: Any) -> Self:
+    def __call__(self, func: Literal[None] = None, **override_settings: Any) -> Self:
         ...
 
-    def __call__(self, func=None, **override_settings: Any):
+    def __call__(self, func: Callable[P, T] | None = None, **override_settings: Any) -> Self | Callable[P, T] | None:
```

Comment:
Good catch, it's a leftover from previous version -- note the change below on line 153, previously this method would return `None` when no parameters were passed.